harness/harness-core
# [fix]: [pl-28198] HashiCorpVault - Throw an exception if secret delet…

…ion fails

## Describe your changes

This PR handles the `HashiCorpVaultRuntimeException` that is thrown while deleting a secret from the HashiCorp Vault.

While deleting the secret in HashiCorp vault if HashiCorpVaultRuntimeException is thrown due to invalid vault configuration then the handler catches the exception and logs it but doesn’t propagate it back to the manager. Thus the UI is unaware of the deletion failure and assumes that the deletion of secret succeeded and removes the secret from the UI.

Test Scenario -

1. Create a HashiCorp secret manager connector (with Root token) and add a secret in HashiCorp Vault.

2. Update the HashiCorp secret manager with invalid token which will cause the connection validation to fail.

3. Now delete the previously stored secret in HashiCorp Vault. The delegate secrets management won’t be able to delete the secret due to the invalid token and throws an HashiCorpVaultRuntimeException

4. However, since this exception is just logged and not propagated back to the NG Manager from the delegate. The UI assumes that the secret was deleted and removes it however its not deleted in Hashicorp Vault.

The expected outcome is the secret not being removed from the UI since deletion of secret failed. Thus, rather than logging the exception it needs to be propagated back to the UI.

## Checklist
- [ Yes ] I've documented the changes in the PR description. - [PL-28198](https://harness.atlassian.net/browse/PL-28198) contains the detailed info on the issue.
- [ Yes ] I've tested this change either in PR or local environment. - Tested in local test environment.
- [ N/A ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions


## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/harness/harness-core/37554)
